src/components/ArticleById.jsx

```diff
@@ -0,0 +1,38 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { getSingleArticle } from "../api";
+import { useNavigate } from "react-router-dom";
+
+function ArticleById() {
+  const { article_id } = useParams();
```

Comment:
we need some loading state here to let the user know the page is loading and there isnt an issue just the internet being slow!